### ISSUE TITLE
False positive: B018

### ISSUE THREAD

**Issue by wmortal** [astral-sh/ruff#9082]
Ruff 0.1.7
pyproject.toml:
```
[tool.ruff.lint]
extend-select = ["B"]  # flake8-bugbear
```
Command:
`ruff check /tmp/ruff/example.py`

Source:
```
class Example:
    @property
    def name(self) -> str:
        return "example"


example = Example()
example.name
^^^^^^^^^^^^
B018 Found useless expression. Either assign it to a variable or remove it

with pytest.raises(ValueError):
    example.name
    ^^^^^^^^^^^^
    B018 Found useless expression. Either assign it to a variable or remove it

```

I never encountered that error in a production code, but I have a lot of it in my tests, that check if calling that property raises an error (second example of the error). Similarly to what was described in #3831. I understand that this is context-specific and in some situations it would be better to get that error even for the cost of a few false-positives. Or maybe it could be less strict in a context of test files? Removing the _property_ decorator and calling it as a method fixes the error.
